src/openrct2/entity/Staff.cpp

```diff
@@ -2288,7 +2289,8 @@ bool Staff::UpdateFixingMoveToStationStart(bool firstRun, const Ride& ride)
 {
     if (!firstRun)
     {
-        if (ride.GetRideTypeDescriptor().HasFlag(RIDE_TYPE_FLAG_HAS_SINGLE_PIECE_STATION | RIDE_TYPE_FLAG_HAS_NO_TRACK))
+        if (ride.GetRideTypeDescriptor().HasFlag(RIDE_TYPE_FLAG_HAS_SINGLE_PIECE_STATION)
+            && !ride.GetRideTypeDescriptor().HasFlag(RIDE_TYPE_FLAG_HAS_TRACK))
```

Comment:
I believe this is not the same as before, the query before is "single piece station OR no tracks" now its "single piece station AND no tracks", this is why the tests are failing since there is a difference in staff behavior now.

Reply:
But that's a bitwise OR, which binds the two flags together, so the `HasFlags()` method will only return `true` if both flags are set, isn't that right?

I can double check that in a while.

Reply:
No thats not correct, it returns true if the (value & flags) != 0

Reply:
You're absolutely right. I'm fixing that.